drivers/console/aarch64/console.S

```diff
@@ -95,10 +95,15 @@ endfunc console_getc
 	 * data that hasn't been output. It returns 0
 	 * upon successful completion, otherwise it
 	 * returns -1.
-	 * Clobber list : x0, x1
+	 * Clobber list : x0, x1, x2, x3, x4
 	 * ---------------------------------------------
 	 */
 func console_flush
+#if COREBOOT
+	mov	x4, x30
+	bl	coreboot_cbmc_flush
+	mov	x30, x4
+#endif
```

Comment:
For the crash console, wouldn't it be possible for the coreboot console to allocate some scratch space for its `plat_crash_console_putc()` implementation to use instead of registers? Any initialization needed can be done within `plat_crash_console_init()`.

I see the general idea of providing branch points at the top level for different console implementations, however I don't think this is very scalable or maintainable since it distributes console specific functionality across several places in generic code. You've already had to plug into 3 different places in generic code to make this work and more may be needed in future.

I don't really understand this comment:

> I feel that the decision which consoles drivers to use shouldn't really be hidden in the platform...

The general philosophy in TF is that platform code is precisely where variations like this should be implemented. We want to work towards reducing build flags in generic code, not proliferating them.